kiliautoml/models/_detectron2_semantic_segmentation.py

```diff
@@ -0,0 +1,520 @@
+            record["annotations"] = objs
+            dataset_dicts.append(record)
+
+        return dataset_dicts
+
+    def train(
+        self,
+        *,
+        assets: List[AssetT],
+        label_merge_strategy: LabelMergeStrategyT,
+        epochs: int,
+        batch_size: int,
+        clear_dataset_cache: bool,
+        disable_wandb: bool,
+        verbose: int,
+        api_key: str,
+    ):
+        """Download Kili assets, convert to coco format, then to detectron2 format, train model."""
+        _ = verbose
+        _ = disable_wandb
+        if not disable_wandb:
+            kili_print("Wandb is not yet available on Detectron2. But tensorboard is available.")
+        _ = label_merge_strategy
+
+        model_path_repository_dir = Path.model_repository_dir(
+            HOME, self.project_id, self.job_name, self.model_repository
+        )
+        if clear_dataset_cache:
+            shutil.rmtree(model_path_repository_dir, ignore_errors=True)
+        model_dir = PathDetectron2.append_model_dir(model_path_repository_dir)
+        data_dir = PathDetectron2.append_data_dir(model_path_repository_dir)
+        eval_dir = PathDetectron2.append_output_evaluation(model_path_repository_dir)
+
+        # 1. Convert to COCO format
+        _, classes = convert_kili_semantic_to_coco(
+            job_name=self.job_name, assets=assets, output_dir=data_dir, api_key=api_key
+        )
+
+        # 2. Convert to Detectron2 format
+        MetadataCatalog.clear()
+        DatasetCatalog.clear()
+        for d in ["train", "val"]:
+            # TODO: separate train and test
+            DatasetCatalog.register("dataset_" + d, lambda d=d: self.get_coco_dicts(data_dir))
+            MetadataCatalog.get("dataset_" + d).set(thing_classes=classes)
+
+        # 3. Train model
+        cfg = self.get_cfg_kili(assets, epochs, batch_size, model_dir, classes)
+        trainer = DefaultTrainer(cfg)
+        trainer.resume_or_load(resume=False)
+        res = trainer.train()
+        kili_print("Training metrics", res)
+
+        # 4. Inference
+        # Inference should use the config with parameters that are used in training
+        # cfg now already contains everything we've set previously. We changed it a little bit for inference: # noqa
+        cfg.MODEL.WEIGHTS = os.path.join(
+            cfg.OUTPUT_DIR, "model_final.pth"
+        )  # path to the model we just trained
+        cfg.MODEL.ROI_HEADS.SCORE_THRESH_TEST = 0.3  # set a custom testing threshold
+        predictor = DefaultPredictor(cfg)
+
+        evaluator = COCOEvaluator("dataset_val", output_dir=eval_dir)
+        val_loader = build_detection_test_loader(cfg, "dataset_val")  # type:ignore
+        eval_res = inference_on_dataset(predictor.model, val_loader, evaluator)
+        kili_print(eval_res)
+        kili_print(f"Evaluations results are available in {eval_dir}")
+        kili_print("The logs and model are saved in ", cfg.OUTPUT_DIR)
+        return eval_res["segm"]["AP"]
```

Comment:
by default train is supposed to return the loss?

Reply:
yes, but there is not a single loss here, and I'm anticipating the future ticket for more interpretable training metrics